### ISSUE TITLE
Add to texture import settings "Pixel Format" option

### ISSUE THREAD

**Issue by Summersay415** [godotengine/godot-proposals#2107]
### Describe the project you are working on
Any

### Describe the problem or limitation you are having in your project
When I am importing textures I can't select pixel format(e.g. "L8, L8A8, RGB24, RGBA32, RGBA4444, RGBA5551, DXT1-DXT5")

### Describe the feature / enhancement and how it helps to overcome the problem or limitation
Add to texture import settings "Pixel Format" option with values "L8, L8A8, RGB24, RGBA32, RGBA4444, RGBA5551, DXT1-DXT5" and add correspond import.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams
Add to texture import settings "Pixel Format" option with values "L8, L8A8, RGB24, RGBA32, RGBA4444, RGBA5551, DXT1-DXT5" and add correspond import.

### If this enhancement will not be used often, can it be worked around with a few lines of script?
This is import function and cannot be realized with script.

### Is there a reason why this should be core and not an add-on in the asset library?
This is import function.

**Comment by Calinou** [comment]
The pixel format is chosen automatically based on the input texture. In the interest of simplicity, there never was a way to force it in the Import dock and I doubt this will change.

**Comment by SIsilicon** [comment]
Then can't that just be an "advanced" option? This can actually be useful for when you have textures that store data that an RGB(A) wouldn't need to.

**Comment by Calinou** [comment]
> This can actually be useful for when you have textures that store data that an RGB(A) wouldn't need to.

In this case, it's better to remove the alpha channel from your original texture to save storage space. There are many tools to do this such as [oxipng](https://github.com/shssoichiro/oxipng).

**Comment by SIsilicon** [comment]
Alpha isn't the only channel that might not need to be omitted. Like I said, a grey scale only needs one channel, and yet can only be imported in 3.

**Comment by Zylann** [comment]
> This is import function and cannot be realized with script.

It can, but it's extremely hard https://github.com/godotengine/godot-proposals/issues/1943

**Comment by Calinou** [comment]
> Alpha isn't the only channel that might not need to be omitted. Like I said, a grey scale only needs one channel, and yet can only be imported in 3.

I double-checked and this used to be correct until Godot 3.4, but not since Godot 3.4. The reason is that since Godot 3.4, lossless WebP is used instead of PNG to provide smaller file sizes for lossless-compressed files. However, we don't make use of WebP's grayscale formats to reduce memory usage (or at least perform a load-time format conversion when possible).
This can be worked around by enabling **Force Png** in the Project Settings, selecting all your textures in the Import dock then clicking **Reimport**. Double-click them and you should see the correct format appear in the inspector preview.

I've tested dozens of formats with all compression modes on `master` 1032c2c43, and here are my findings with the default project settings (= lossless WebP is used):

- **VRAM Uncompressed** mode (Uncompressed in `3.x`) will use the L8 or LA8 formats when the source format is grayscale or grayscale + alpha. This only applies to PNG and JPEG source images, not BMP, TGA or WebP (despite those formats being able to store grayscale and/or grayscale + alpha).
- **Lossless** mode will only use RGB8 or RGBA8, never L8 or LA8.
- **Lossy** mode will only use RGB8 or RGBA8, never L8 or LA8.
- **VRAM Compressed** mode will use DXT1 compression for RGB8 textures and DXT5 compression for RGBA8 textures. This is generally what's expected, so the current behavior seems fine to me.
- **Basis Universal** mode (new in `master`) uses BPTC_RGBA for all textures, even if no alpha channel is needed. I'm not sure if this is a bug or because the BPTC format is only available in RGBA form (not RGB).
- TGA in LA8 format is not supported by Godot, despite the format supporting this. BMP does not have a LA8 format available.

**Testing project (`master`):** [test_image_formats.zip](https://github.com/godotengine/godot-proposals/files/7815684/test_image_formats.zip)

**Comment by viktor-ferenczi** [comment]
I have this same problem while trying to load images into a Texture2DArray, where all layers must have the exact same format. Layer 0 needs to be full black, but the automatic import screws it by forcing DXT1 RGB8 instead of DXT5 RGBA8 the other layers have. It results in a "All images must share the same format" error. I'm aware it may be worked around, but why to work on that if the setup could be very simple by manual format selection?

**Comment by Zylann** [comment]
How is texture import related to loading *Images* into a Texture2DArray? An `Image` is not a texture, and can be loaded with the `Image.load` function. You won't have any issue with them being compressed that way. Are you hitting this issue because you did `load("some_texture.png").get_data()`? If you did, you should likely not load the images this way I guess.

**Comment by viktor-ferenczi** [comment]
You are right, I mess up something in code for sure. Just working on fixing it. But still, being able to explicitly specify an import format would still be nice.
